jwt.go

```diff
@@ -125,21 +154,48 @@ func ParsePrivateKey(data []byte) (interface{}, error) {
 	return rsaKey, nil
 }
 
-// GetTokenFromRequest takes the first Authorization header and extracts the bearer json web token
-func GetTokenFromRequest(r *http.Request) (string, error) {
-	authHeader := r.Header.Get("Authorization")
-	parts := strings.Split(authHeader, " ")
-	if len(parts) != 2 {
-		return "", errors.New("no valid authorization header")
+// GetTokenFromRequest takes the first Authorization header or `token` GET pararm , then
+// extract the token prefix and json web token
+func GetTokenFromRequest(r *http.Request) (prefix string, token string, err error) {
+
+	tokenList, ok := r.Header[AuthorizationHeader]
+	// pull from GET if not in the headers
+	if !ok || len(tokenList) < 1 {
+		tokenList, ok = r.URL.Query()["token"]
+		prefix = "GET"
+	}
+
+	if len(tokenList) < 1 {
+		prefix = ""
+		return prefix, token, errors.New("no valid authorization header")
+	}
+
+	tokenParts := strings.Fields(tokenList[0])
+	switch len(tokenParts) {
+	case 1:
+		token = tokenParts[0]
+	case 2:
+		prefix = tokenParts[0]
+		token = tokenParts[1]
+	default:
+		return prefix, token, errors.New("invalid token: unexpected number of parts")
 	}
-	return parts[1], nil
+
+	return prefix, token, nil
 }
 
 // GetClaimsFromRequest extracts and validates the token from a request, returning the claims
-func GetClaimsFromRequest(r *http.Request, key interface{}) (Claims, error) {
-	token, err := GetTokenFromRequest(r)
+func GetClaimsFromRequest(r *http.Request, key interface{}, skipValidate bool) (prefix string, claims Claims, err error) {
```

Comment:
perhaps it would be better readable to use a variadic skipValidate parameter so that it can be left out in the default case

Reply:
Seems like an abuse of variadic options

Reply:
yes, a bit. But now there would be in 99% of all cases a trailing "false" in the parameter list. Just a question of style, this will not prevent us from merging this PR

Reply:
true, i figured that was just the price we pay for using Go 😄

Reply:
Personally I always try to avoid function signatures with bool flag arguments. In most cases I think it's clearer from an api perspective to just have a separate function, e.g. `GetClaimsFromRequestWithValidation`

Reply:
I just pushed a version of the code that looks like this, for review.  If we don't like it we can revert it :)